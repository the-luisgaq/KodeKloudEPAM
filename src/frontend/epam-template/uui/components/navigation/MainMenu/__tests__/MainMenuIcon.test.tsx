import React from 'react';
import { renderSnapshotWithContextAsync } from '@epam/uui-test-utils';
import { MainMenuIcon } from '../MainMenuIcon';
import { ReactComponent as CalendarIcon } from '@epam/assets/icons/action-calendar-fill.svg';

describe('MainMenuIcon', () => {
    it('should be rendered correctly with minimum props', async () => {
        const tree = await renderSnapshotWithContextAsync(<MainMenuIcon icon={ CalendarIcon } />);
        expect(tree).toMatchSnapshot();
    });

    it('should be rendered correctly', async () => {
        const tree = await renderSnapshotWithContextAsync(
            <MainMenuIcon
                icon={ CalendarIcon }
                target="_blank"
                link={ { pathname: '/' } }
                collapseToMore
                estimatedWidth={ 120 }
                priority={ 6 }
            />,
        );
        expect(tree).toMatchSnapshot();
    });
});
