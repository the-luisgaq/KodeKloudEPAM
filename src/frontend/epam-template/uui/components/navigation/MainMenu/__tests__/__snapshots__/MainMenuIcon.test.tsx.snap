// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainMenuIcon should be rendered correctly 1`] = `
<DocumentFragment>
  <a
    class="uui-button-box uui-enabled -clickable uui-anchor container uui-icon_button uui-color-neutral root container"
    href=""
    rel="noopener noreferrer"
    tabindex="0"
    target="_blank"
  >
    <div
      class="container uui-icon uui-enabled"
    >
      <svg
        class=""
      />
    </div>
  </a>
</DocumentFragment>
`;

exports[`MainMenuIcon should be rendered correctly with minimum props 1`] = `
<DocumentFragment>
  <button
    class="uui-button-box uui-enabled container uui-icon_button uui-color-neutral root container"
    tabindex="-1"
    type="button"
  >
    <div
      class="container uui-icon uui-enabled"
    >
      <svg
        class=""
      />
    </div>
  </button>
</DocumentFragment>
`;
